src/nupic/algorithms/Anomaly.hpp

```diff
@@ -112,10 +113,16 @@ namespace nupic
                        const std::vector<UInt>& predicted,
                        Real64 inputValue=0, UInt timestamp=0);
 
+//helper
+        void setLikelihood(AnomalyLikelihood newInstance) {
+          this->likelihood_ = newInstance;
+        }
+
       private:
         AnomalyMode mode_;
         Real32 binaryThreshold_;
         std::unique_ptr<nupic::util::MovingAverage> movingAverage_;
+        AnomalyLikelihood likelihood_; //TODO which params/how pass them to constructor?
```

Comment:
Is it worth adding the code to instantiate only if needed and destruct, rather than just make an instance and use it when needed?